pkg/resourcemanager/controller/csrapprover/reconciler.go

```diff
@@ -197,25 +198,31 @@ func (r *Reconciler) mustApproveKubeletServing(ctx context.Context, csr *certifi
 
 	var (
 		hostNames   []string
-		ipAddresses []string
+		ipAddresses []netip.Addr
 	)
 
 	for _, address := range node.Status.Addresses {
-		if address.Type == corev1.NodeHostName || address.Type == corev1.NodeInternalDNS || address.Type == corev1.NodeExternalDNS {
+		switch address.Type {
+		case corev1.NodeHostName, corev1.NodeInternalDNS, corev1.NodeExternalDNS:
 			hostNames = append(hostNames, address.Address)
-		}
-		if address.Type == corev1.NodeInternalIP || address.Type == corev1.NodeExternalIP {
-			ipAddresses = append(ipAddresses, address.Address)
+		case corev1.NodeInternalIP, corev1.NodeExternalIP:
+			comparableIP, err := netip.ParseAddr(address.Address)
+			if err != nil {
+				return fmt.Sprintf("IP address %q in node.Status.Addresses is invalid: %v", address.Address, err), false, nil //nolint:nilerr
```

Comment:
Is this not rather an actual error similar to for example line 192.

```suggestion
				return "", false, fmt.Errorf("failed to parse IP address %q in node.Status.Addresses: %w", address.Address, err)
```

Reply:
If we return the error here, instead of `return "deny message", false, nil` will that deny the CSR?

Reply:
> If we return the error here, instead of `return "deny message", false, nil` will that deny the CSR?

Nope, it won't. If you return an error, reconciliation will start again and nothing would happen to the CSR.
IMO your initial solution is correct, as this is a permanent error and the CSR should be denied. Line 192 is a temporary error.

Reply:
Good point. Thanks for clarifying.